src/Type/Traits/UndecidedComparisonCompoundTypeTrait.php

```diff
@@ -13,11 +13,19 @@ trait UndecidedComparisonCompoundTypeTrait
 
 	public function isGreaterThan(Type $otherType, PhpVersion $phpVersion): TrinaryLogic
 	{
+		if ($otherType->isNull()->yes() && $this->isObject()->yes()) {
```

Comment:
This add some logic in the trait but I didn't find another way except creating an ObjectComparisonCompoundTypeTrait and an ObjectComparisonTypeTrait.

Do you prefer to create two new traits ?